core/dbt/graph/selector_methods.py

```diff
@@ -527,14 +513,21 @@ def search(
 
         state_checks = {
             # it's new if there is no old version
-            'new': lambda old, _: old is None,
+            'new':
+                lambda old, _: old is None,
             # use methods defined above to compare properties of old + new
-            'modified': self.check_modified,
-            'modified.body': self.check_modified_body,
-            'modified.configs': self.check_modified_configs,
-            'modified.persisted_descriptions': self.check_modified_persisted_descriptions,
-            'modified.relation': self.check_modified_relation,
-            'modified.macros': self.check_modified_macros,
+            'modified':
+                self.check_modified_content,
+            'modified.body':
+                self.check_modified_factory('same_body'),
```

Comment:
Let me know if you think we should go change the underlying function names instead of using the factory here! they are used somewhere but I think it is not too much

Reply:
I like the factory. It avoids duplicating the code in the functions.